src/aspire/denoise/covar2d.py

```diff
@@ -33,12 +45,60 @@ def get_covar(self, coeffs=None, mean_coeff=None,  do_refl=false):
         :return: The covariance matrix of coefficients for all images.
 
         """
-        pass
+        if coeffs is None:
+            raise RuntimeError('The coefficients need to be calculated!')
```

Comment:
Why have a default value on an argument that you're never going to allow anyway?

Reply:
Originally thought that we can call the evaluate function here to get coefficients. This will be implemented later. Fixed in the commit.